commit db9183fc34cfc0cf977bd2fe4aa2bd555c8e8066 (refs/changes/79/1579/3)
Author: Trent Mick <trentm@gmail.com>
Date:   2017-02-27T11:06:18-08:00 (2 years, 7 months ago)
    
    DOCKER-1006 jenkins TAP parser doesn't like sdc-docker's cli-packagelabels.test.js output
    Reviewed by: Todd Whiteman <todd.whiteman@joyent.com>
    Approved by: Todd Whiteman <todd.whiteman@joyent.com>

diff --git a/test/integration/cli-packagelabels.test.js b/test/integration/cli-packagelabels.test.js
index 7253735..d296275 100644
--- a/test/integration/cli-packagelabels.test.js
+++ b/test/integration/cli-packagelabels.test.js
@@ -40,19 +40,28 @@ var papi;
 // --- Disable these tests if too old
 
 if (cliVersion) {
+    /*
+     * We don't emit content to stdout because strict TAP parsing must
+     * see the "TAP version 13" line of output before "# comments ...".
+     * Eventually <https://github.com/substack/tape/pull/197> would allow
+     * `opts.skip` to be a string reason included in the TAP output.
+     */
     cliVersion = cliVersion.split('.')[0] + '.' + cliVersion.split('.')[1];
     if (common.apiVersionCmp(cliVersion, 1.8) < 0) {
-        console.log('# disabling test, cli version is: '
-            + process.env.DOCKER_CLI_VERSION);
         opts.skip = true;
-    } else {
-        console.log('# cli version is: ' + process.env.DOCKER_CLI_VERSION);
     }
 }
 
 
 // --- Tests
 
+test('note if we are skipping due to old CLI version', function (tt) {
+    if (opts.skip) {
+        tt.comment('skipping tests because CLI version is < 1.8: it is '
+            + process.env.DOCKER_CLI_VERSION);
+    }
+    tt.end();
+});
 
 test('setup docker environment/cli', opts, function (tt) {
     h.createPapiClient(function (err, _papi) {
